rl_studio/agents/f1/utils.py

```diff
@@ -0,0 +1,81 @@
+import pickle
+import datetime
+from agents.f1 import settings
+
+
+#TODO Since these utils are algorithm specific, those should stay in the algorithm folder somehow tied to its algorithm class
+
+
+def load_model(params, qlearn, file_name):
+
+    qlearn_file = open("./logs/qlearn_models/" + file_name)
+    model = pickle.load(qlearn_file)
+
+    qlearn.q = model
+    qlearn.alpha = params.algorithm["params"]["alpha"]
+    qlearn.gamma = params.algorithm["params"]["epsilon"]
+    qlearn.epsilon = params.algorithm["params"]["gamma"]
+
+    # highest_reward = settings.algorithm_params["highest_reward"]
+
+    print(f"\n\nMODEL LOADED. Number of (action, state): {len(model)}")
+    print(f"    - Loading:    {file_name}")
+    print(f"    - Model size: {len(qlearn.q)}")
+    print(f"    - Action set: {settings.actions_set}")
+    print(f"    - Epsilon:    {qlearn.epsilon}")
+    print(f"    - Start:      {datetime.datetime.now()}")
+
+
+def save_model(qlearn, current_time, states, states_counter, states_rewards):
+    # Tabular RL: Tabular Q-learning basically stores the policy (Q-values) of  the agent into a matrix of shape
+    # (S x A), where s are all states, a are all the possible actions. After the environment is solved, just save this
+    # matrix as a csv file. I have a quick implementation of this on my GitHub under Reinforcement Learning.
+
+    # Q TABLE
+    base_file_name = "_act_set_{}_epsilon_{}".format(
+        settings.qlearn.actions_set, round(qlearn.epsilon, 2)
+    )
+    file_dump = open(
+        "./logs/qlearn_models/1_" + current_time + base_file_name + "_QTABLE.pkl", "wb"
+    )
+    pickle.dump(qlearn.q, file_dump)
+    # STATES COUNTER
+    states_counter_file_name = base_file_name + "_STATES_COUNTER.pkl"
+    file_dump = open(
+        "./logs/qlearn_models/2_" + current_time + states_counter_file_name, "wb"
+    )
+    pickle.dump(states_counter, file_dump)
+    # STATES CUMULATED REWARD
+    states_cum_reward_file_name = base_file_name + "_STATES_CUM_REWARD.pkl"
+    file_dump = open(
+        "./logs/qlearn_models/3_" + current_time + states_cum_reward_file_name, "wb"
+    )
+    pickle.dump(states_rewards, file_dump)
+    # STATES
+    steps = base_file_name + "_STATES_STEPS.pkl"
+    file_dump = open("./logs/qlearn_models/4_" + current_time + steps, "wb")
+    pickle.dump(states, file_dump)
```

Comment:
I  think we can change this strings by f-strings:

```python
 file_dump = open(
      "./logs/qlearn_models/3_" + current_time + states_cum_reward_file_name, "wb"
  )
```

f-string:

```bash
 file_dump = open(f"./logs/qlearn_models/3_{current_time}{states_cum_reward_file_name},wb")
```

Reply:
I didn´t know them. Thanks!!! :) I change them today